test/markdown-link-check.test.js

```diff
@@ -96,7 +97,7 @@ describe('markdown-link-check', function () {
                 httpHeaders: [
                     {
                         urls: [baseUrl + '/basic-auth'],
-                        headers: { 'Authorization': 'Basic Zm9vOmJhcg==', 'Foo': 'Bar' }
+                        headers: { 'Authorization': 'Basic {{env.BASIC_AUTH_TOKEN}}', 'Foo': 'Bar' }
```

Comment:
Be careful that you are testing 2 things here at the same time: token replacement and auth headers.
Maybe splitting the test by keeping the old one as is and creating a new one "should replace token by env var in headers". Your token system can be used for anything in headers and not only auth. It will be the main usecase for sure, but you wrote something more generic IMO.

Reply:
You're right, I've splitted them up as requested. See [here]( https://github.com/tcort/markdown-link-check/pull/146/files#diff-71f4f6afd52d6343234f94db195fef87065ba6740798f994791ecded7b198925R253-R299
).